modin/engines/base/frame/data.py

```diff
@@ -822,11 +822,25 @@ def _map(self, func, dtypes=None):
             dtypes = pandas.Series(
                 [np.dtype(dtypes)] * len(self.columns), index=self.columns
             )
+
+        index_validation_required = (
```

Comment:
There is possibly a better way to do this with an argument to `_map` instead of this detection, because it can be used more generally than just if the first partitions do not have the same length.

This also forces a blocking call which is undesirable for future query planning work.

Reply:
done